model/antity/userAntity.php

```diff
@@ -0,0 +1,134 @@
+<?php
+
+Class User {
+
+    protected $id;
+    protected $firstname;
+    protected $lastname;
+    protected $email;
+    protected $phone;
+    protected $password;
+    protected $role;
+
+    public function __construct(array $data){
+        $this->dbConnect();
+        $this->hydrate($data);
+    }
+
+    public function hydrate(array $data)
+    {
+      foreach ($donnees as $key => $value)
+      {
+        $method = 'set'.ucfirst($key);
+        
+        if (method_exists($this, $method))
+        {
+          $this->$method($value);
+          
+        }
+      }
+    }
+
+    // Setteur
+    public function setId($id) {
+        $id = (int) $id;
+
+        if($id > 0) {
+            $this->_id = $id;
+        }
+    }
+
+    
+    public function setFirstname($firstname) {
+        if($firstname = (string)$firstname)
+        {
+            $this->_firstname = $firstname;
```

Comment:
Ton attribut de classe s'appelle $firstname. C'est donc $this->firstname et non pas $this->_firstname

Valable pour tes autres méthodes.